adafruit_tlc5947.py

```diff
@@ -226,20 +239,22 @@ def create_pwm_out(self, channel):
     # like when using the PWMOut mock class).
     def __len__(self):
         """Retrieve the total number of PWM channels available."""
-        return 24  # Always 24 channels on the chip.
+        return _CHANNELS * self._n  # number channels times number chips.
 
     def __getitem__(self, key):
-        """Retrieve the 12-bit PWM value for the specified channel (0-23).
+        """Retrieve the 12-bit PWM value for the specified channel (0-max).
+        max depends on the number of boards.
         """
-        assert 0 <= key <= 23
+        assert 0 <= key < _CHANNELS * self._n
```

Comment:
To allow for Python's negative indices (adressing elements from the rear end of an array), the suggested change is needed. Otherwise the negative index adressing near the end of tlc5947_chain.py would fail.
```suggestion
        if key < 0:      # allow reverse adressing with negative index
            key = key + _CHANNELS * self._n
        assert 0 <= key < _CHANNELS * self._n
```